extensions/amp-story/1.0/amp-story-request-service.js

```diff
@@ -55,13 +55,30 @@ export class AmpStoryRequestService {
 
   /**
    * Retrieves the publisher share providers.
-   * Has to be called through `loadShareConfig`.
+   * @param  {!Element} shareConfigEl
    * @return {(!Promise<!JsonObject>|!Promise<null>)}
    */
-  loadShareConfigImpl_() {
-    const shareConfigEl = this.storyElement_.querySelector(
-      'amp-story-social-share, amp-story-bookend'
-    );
+  getInlineConfig(shareConfigEl) {
+    try {
+      return Promise.resolve(getChildJsonConfig(shareConfigEl));
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  }
+
+  /**
+   * Retrieves the publisher share providers.
+   * @param  {?Element} element
+   * @return {(!Promise<!JsonObject>|!Promise<null>)}
+   * @public
+   */
+  loadConfigImpl(element) {
+    const shareConfigEl = element
+      ? element
+      : this.storyElement_.querySelector(
+          'amp-story-social-share, amp-story-bookend'
+        );
```

Comment:
I don't think we should actually default to the `amp-story-social-share` element, since it seems like calling `loadConfigImpl()` for the shopping config (or any future configs) could result in the method operating on the share config instead. I think a private `loadShareConfigImpl_()`  would resolve this by allowing you to separate the share config logic from the loading logic. The end result would look something like this:

loadShareConfigImpl_() {
  const shareConfigEl = this.storyElement_.querySelector(
    'amp-story-social-share, amp-story-bookend'
  );
  return this.loadConfigImpl(shareConfigEl);
}

loadConfigImpl(element) {
  if (!element) {
    return Promise.resolve();
  }
  ...
}

Reply:
I have done some further refactoring of the code and have made it so that the query selection is done before the call to loadconfigImpl.